network.cpp

```diff
@@ -189,6 +224,44 @@ bool net_behaves_like_the_function(const network &n, const std::function<double(
 
 }
 
+bool all_weigths_are_active(const network &n)
+{
+  std::vector<std::vector<std::vector<weight>>> weights = n.get_net_weights();
+
+  for(auto &layer : weights ){
+      for(auto &node : layer){
+          for (auto &weight : node){
+              if(!weight.is_active()){
+                  return false;
+                }
+            }
+        }
+    }
+  return true;
+}
+
+bool on_average_an_nth_of_the_activations_are_mutated(const network &n, const std::vector<weight>&registered_mutations,
+                                                      const double &mut_rate, int repeats)
+{
+  int number_of_weights = get_number_weights(n);
+  size_t number_mutations = registered_mutations.size();
+  double error = 0.1 * repeats;
+  return !(number_mutations < mut_rate * repeats * number_of_weights - error ||
+           number_mutations > mut_rate * repeats * number_of_weights + error);
```

Comment:
mmmm I think this is not what I intended for the function to do. 
It is a bit hard to read what the function is doing but all in all I gather that what you are doing is to check that the right proportion of weights has mutated.
i think this works for our test, but it is higly dependent on the condition of our test( all weight are active).
What I wanted to do here is just to check that on average  an 1/n proportion of the weights in a network is active.
Can you try and change the function so that it performs this  (simpler) task?

Reply:
Okay yes, but then should it be 1/n proportion of the weights in a network is inactive ? (I mean i know they're easy to go from one to the other, but I mean in the assert ? Because the networks are initialized as fully active, so all mutations will switch the weights to inactive